### ISSUE TITLE
ActivityBot Display Calibration interpolation table in demo - new display issue

### ISSUE THREAD

**Issue by AndyLindsay** [parallaxinc/BlocklyProp#1008]
**Output from Blockly Demo:**

Connection established with: COM245 using baud rate 115200

=== LEFT SERVO ===
Table Entries = 0, Zero Speed Index = 0

Index, Servo Drive, Encoder Ticks/Second

=== RIGHT SERVO ===
Table Entries = 0, Zero Speed Index = 0

Index, Servo Drive, Encoder Ticks/Second

**Output from SimpleIDE**

=== LEFT SERVO ===
Table Entries = 89, Zero Speed Index = 41

Index, Servo Drive, Encoder Ticks/Second
0, -194, 1000
1, -189, 179
2, -184, 177
3, -179, 181
4, -174, 181
5, -169, 177
6, -164, 175
7, -159, 175
8, -154, 179
9, -149, 174
10, -144, 172
11, -139, 172
12, -134, 163
13, -129, 170
14, -124, 166
15, -119, 161
16, -114, 160
17, -109, 154
18, -104, 150
19, -99, 151
20, -94, 146
21, -89, 139
22, -84, 133
23, -79, 128
24, -74, 124
25, -69, 118
26, -64, 111
27, -59, 100
28, -54, 93
29, -49, 85
30, -44, 78
31, -39, 69
32, -34, 60
33, -29, 53
34, -24, 42
35, -19, 34
36, -14, 26
37, -9, 17
38, -7, 13
39, -5, 9
40, -3, 7
41, 5, 0
42, 13, 7
43, 15, 12
44, 17, 15
45, 19, 18
46, 21, 22
47, 26, 31
48, 31, 40
49, 36, 50
50, 41, 57
51, 46, 66
52, 51, 74
53, 56, 83
54, 61, 89
55, 66, 95
56, 71, 102
57, 76, 110
58, 81, 120
59, 86, 126
60, 91, 130
61, 96, 138
62, 101, 143
63, 106, 143
64, 111, 152
65, 116, 148
66, 121, 152
67, 126, 154
68, 131, 158
69, 136, 160
70, 141, 163
71, 146, 166
72, 151, 163
73, 156, 163
74, 161, 168
75, 166, 170
76, 171, 170
77, 176, 170
78, 181, 170
79, 186, 172
80, 191, 172
81, 196, 170
82, 201, 172
83, 176, 170
84, 181, 170
85, 186, 172
86, 191, 172
87, 196, 170
88, 201, 1000

=== RIGHT SERVO ===
Table Entries = 89, Zero Speed Index = 40

Index, Servo Drive, Encoder Ticks/Second
0, -195, 1000
1, -190, 177
2, -185, 173
3, -180, 175
4, -175, 175
5, -170, 173
6, -165, 177
7, -160, 175
8, -155, 171
9, -150, 171
10, -145, 169
11, -140, 171
12, -135, 168
13, -130, 164
14, -125, 163
15, -120, 158
16, -115, 161
17, -110, 154
18, -105, 156
19, -100, 149
20, -95, 145
21, -90, 145
22, -85, 139
23, -80, 133
24, -75, 131
25, -70, 121
26, -65, 116
27, -60, 107
28, -55, 103
29, -50, 93
30, -45, 88
31, -40, 76
32, -35, 69
33, -30, 58
34, -25, 50
35, -20, 41
36, -15, 31
37, -10, 22
38, -5, 12
39, -3, 8
40, 6, 0
41, 15, 8
42, 17, 12
43, 19, 16
44, 21, 19
45, 23, 22
46, 28, 33
47, 33, 40
48, 38, 50
49, 43, 61
50, 48, 68
51, 53, 79
52, 58, 85
53, 63, 95
54, 68, 101
55, 73, 114
56, 78, 117
57, 83, 123
58, 88, 129
59, 93, 129
60, 98, 139
61, 103, 144
62, 108, 147
63, 113, 153
64, 118, 155
65, 123, 152
66, 128, 153
67, 133, 158
68, 138, 161
69, 143, 161
70, 148, 164
71, 153, 163
72, 158, 166
73, 163, 166
74, 168, 169
75, 173, 171
76, 178, 168
77, 183, 164
78, 188, 173
79, 193, 175
80, 198, 175
81, 203, 171
82, 173, 171
83, 178, 168
84, 183, 164
85, 188, 173
86, 193, 175
87, 198, 175
88, 203, 1000

**Comment by MatzElectronics** [comment]
Okay - this needs to be looked at specifically (and globally) - it stinks,
but the table was near useless in it's current state because of scrolling
issues with the terminal itself.  I'll take this on as part of a larger
terminal fix.

*Matthew Matz  |  STEM/Robotics Educator*
Parallax Inc.  |  Direct: 916-625-3019  |  www.parallax.com  |  @M
atzElectronics <http://twitter.com/MatzElectronics>

**Comment by mrodriguez-parallax** [comment]
I'm getting some strange values on my interpolation table. Below is part of the output after calibration(I wasn't able to get the entire table). When I run the calibration program using Simple IDE the interpolation table outputs values corresponding to a good calibration. I calibrated two different Activity Bots and both output the same type of data that I included below in item 1. I then went over to Simple IDE and ran the interpolation table program. The values look good as shown in item 2.

**item 1**
Connection established with: COM11 using baud rate 115200

=== LEFT SERVO ===
Table Entries = 88, Zero Speed Index = 41

Index, Servo Drive, Encoder Ticks/Second
0, ‑12320962, 10945512
1, ‑11665592, 11075754
2, ‑11010222, 11075756
3, ‑10354852, 10813608
4, ‑9699482, 11075751
5, ‑9044112, 10748069
6, ‑8388742, 10813603
7, ‑7733372, 10748067
8, ‑7078002, 10420383
9, ‑6422632, 9765020
10, ‑5767262, 9699479
11, ‑5111892, 8978577
12, ‑4456522, 8126596
13, ‑3801152, 7274615
14, ‑3145782, 6226025
15, ‑2490412, 5177432
16, ‑1835042, **4128837**
17, ‑1179672, 2752563

**item 2**
**These are the values output from the same calibration this time using Simple IDE.**
Table Entries = 89
Zero Speed Index = 42

Index   Servo Drive     Encoder Ticks/Second
-----   -----------     --------------------
0       -194            1000
1       -189            166
2       -184            166
3       -179            165
4       -174            162
5       -169            166
6       -164            164
7       -159            162
8       -154            163
9       -149            163
10      -144            165
11      -139            162
12      -134            160
13      -129            160
14      -124            158
15      -119            155
16      -114            155
17      -109            152
18      -104            152
19      -99             148
20      -94             145
21      -89             142
22      -84             140
23      -79             134
24      -74             130
25      -69             123
26      -64             115
27      -59             108
28      -54             100
29      -49             92
30      -44             85
31      -39             77
32      -34             69
33      -29             58
34      -24             49
35      -19             41
36      -14             30
37      -9              22
38      -4              13
39      -2              9
40      0               5
41      2               5
42      9               0
43      16              9
44      18              12
45      20              16
46      22              20
47      24              24
48      29              33
49      34              41
50      39              52
51      44              60
52      49              71
53      54              79
54      59              85
55      64              95
56      69              99
57      74              107
58      79              115
59      84              123
60      89              126
61      94              131
62      99              133
63      104             142
64      109             147
65      114             150
66      119             148
67      124             151
68      129             153
69      134             156
70      139             158
71      144             156
72      149             161
73      154             160
74      159             158
75      164             158
76      169             162
77      174             163
78      179             165
79      184             163
80      189             160
81      194             165
82      199             160
83      204             163
84      184             163
85      189             160
86      194             165
87      199             160
88      204             1000


=== RIGHT SERVO ===

Table Entries = 89
Zero Speed Index = 41

Index   Servo Drive     Encoder Ticks/Second
-----   -----------     --------------------
0       -195            1000
1       -190            164
2       -185            164
3       -180            162
4       -175            162
5       -170            161
6       -165            157
7       -160            163
8       -155            157
9       -150            164
10      -145            161
11      -140            161
12      -135            156
13      -130            156
14      -125            153
15      -120            154
16      -115            150
17      -110            146
18      -105            144
19      -100            136
20      -95             135
21      -90             134
22      -85             124
23      -80             125
24      -75             114
25      -70             110
26      -65             103
27      -60             92
28      -55             89
29      -50             77
30      -45             67
31      -40             61
32      -35             51
33      -30             44
34      -25             33
35      -20             23
36      -15             13
37      -13             9
38      -11             9
39      -9              6
40      -7              5
41      0               0
42      7               7
43      9               11
44      11              15
45      13              19
46      15              23
47      20              32
48      25              41
49      30              50
50      35              61
51      40              71
52      45              77
53      50              86
54      55              92
55      60              100
56      65              107
57      70              114
58      75              120
59      80              127
60      85              129
61      90              138
62      95              138
63      100             141
64      105             144
65      110             150
66      115             151
67      120             151
68      125             151
69      130             153
70      135             158
71      140             159
72      145             158
73      150             159
74      155             155
75      160             159
76      165             158
77      170             164
78      175             164
79      180             161
80      185             163
81      190             159
82      195             161
83      200             159
84      180             161
85      185             163
86      190             159
87      195             161
88      200             1000
